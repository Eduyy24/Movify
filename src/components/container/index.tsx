import React from 'react';
import {
  SafeAreaView,
  StatusBar,
  StyleSheet,
  useColorScheme,
} from 'react-native';
import type {PropsWithChildren} from 'react';
import {COLORS} from '../../styles';

export const Container = ({children}: PropsWithChildren) => {
  const isDarkMode = useColorScheme() === 'dark';

  const backgroundStyle = {
    backgroundColor: isDarkMode ? COLORS.BROWN : COLORS.BROWN,
  };

  return (
    <SafeAreaView style={styles.container}>
      <StatusBar
        barStyle={isDarkMode ? 'light-content' : 'dark-content'}
        backgroundColor={backgroundStyle.backgroundColor}
      />
      {children}
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: COLORS.WHITE,
  },
});
