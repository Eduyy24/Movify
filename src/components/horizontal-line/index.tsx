import React from 'react';
import {StyleSheet, View} from 'react-native';
import {COLORS} from '../../styles';

export const HorizontalLine = () => {
  return <View style={styles.line} />;
};

const styles = StyleSheet.create({
  line: {
    width: '100%',
    borderBottomColor: COLORS.ACCENT,
    borderBottomWidth: 1,
    marginVertical: 5,
  },
});
