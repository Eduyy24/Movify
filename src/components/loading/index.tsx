import React, {FC} from 'react';
import {ActivityIndicator, StyleSheet, View} from 'react-native';
import {COLORS} from '../../styles';

export const Loading: FC = () => (
  <View style={styles.container}>
    <ActivityIndicator size="large" color={COLORS.ACCENT} />
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
