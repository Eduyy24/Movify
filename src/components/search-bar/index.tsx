import React from 'react';
import {Image, StyleSheet, TextInput, View} from 'react-native';
import {COLORS} from '../../styles';

export const SearchBar = () => {
  return (
    <View style={styles.container}>
      <TextInput style={styles.inputText} />
      <Image source={require('../../assets/icons/search.png')} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    height: 60,
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: 10,
  },
  inputText: {
    flex: 1,
    borderBottomColor: COLORS.SECONDARY,
    borderBottomWidth: 1,
    height: 40,
  },
});
