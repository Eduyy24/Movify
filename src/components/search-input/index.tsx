import React, {FC} from 'react';
import {Image, StyleSheet, TextInput, View} from 'react-native';
import {COLORS} from '../../styles';

interface Props {
  onChangeSearch: (text: string) => void;
}

export const SearchInput: FC<Props> = ({onChangeSearch}) => {
  return (
    <View style={styles.container}>
      <TextInput style={styles.inputText} onChangeText={onChangeSearch} />
      <Image source={require('../../assets/icons/search.png')} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    height: 60,
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: 10,
  },
  inputText: {
    flex: 1,
    borderBottomColor: COLORS.SECONDARY,
    borderBottomWidth: 1,
    height: 40,
  },
});
