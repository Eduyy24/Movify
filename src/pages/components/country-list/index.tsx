import React from 'react';

export const CountryList = () => {
  return <></>;
};
