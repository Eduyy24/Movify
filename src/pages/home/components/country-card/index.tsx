import React, {FC} from 'react';
import {
  Linking,
  Platform,
  Pressable,
  StyleSheet,
  Text,
  View,
} from 'react-native';
import {COLORS} from '../../../../styles';
import Countries from 'countries-capitals';

interface Props {
  country: string;
}

export const CountryCard: FC<Props> = ({country}) => {
  const countries = new Countries();
  const countryInfo = countries.byName(country);
  const countryJson = countryInfo.toJson();

  const url = Platform.select({
    ios: 'maps:?q=' + country,
    android: 'geo:?q=' + country,
  });

  const onPressGo = () => {
    console.log('country', country, countryJson);
    Linking.canOpenURL(url ?? '').then(supported => {
      if (supported) {
        return Linking.openURL(url ?? '');
      }
    });
  };

  return (
    <View style={styles.container}>
      <Pressable style={styles.containerGo} onPress={onPressGo}>
        <Text style={styles.txtGO}>GO</Text>
      </Pressable>
      <View style={styles.containerInfo}>
        <Text style={styles.txtName}>{country}</Text>
        <Text style={styles.txtCapital}>{countryJson[0]?.city}</Text>
        <Text style={styles.txtRegion}>{countryJson[0]?.location}</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    minHeight: 120,
    padding: 12,
    backgroundColor: COLORS.CLEAR,
    borderRadius: 12,
    flexDirection: 'row',
    alignItems: 'center',
    elevation: 2,
    shadowOffset: {width: -2, height: 4},
    shadowColor: '#171717',
    shadowOpacity: 0.2,
    shadowRadius: 3,
  },
  containerGo: {
    padding: 8,
    backgroundColor: COLORS.PRIMARY,
    borderRadius: 8,
    marginRight: 12,
  },
  containerInfo: {
    flex: 1,
    height: '100%',
    justifyContent: 'space-evenly',
  },
  txtGO: {
    fontSize: 50,
    color: COLORS.CLEAR,
  },
  txtName: {
    fontSize: 24,
    fontWeight: 'bold',
    color: COLORS.BROWN,
  },
  txtCapital: {
    fontSize: 16,
    color: COLORS.BROWN,
  },
  txtRegion: {fontSize: 20, color: COLORS.ACCENT},
});
