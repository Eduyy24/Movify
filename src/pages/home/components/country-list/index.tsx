import React from 'react';
import {useGeneralContext} from '../../../../hooks/useGeneralContext';
import {FlatList, View, Text} from 'react-native';

export const CountryList = () => {
  const {countryList} = useGeneralContext();
  return (
    <View>
      <FlatList
        data={countryList}
        renderItem={({item}) => <Text>{item}</Text>}
      />
    </View>
  );
};
