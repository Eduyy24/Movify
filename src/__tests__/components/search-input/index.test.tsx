import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react-native';
import {SearchInput} from '../../../components/search-input';

describe('<SearchInput />', () => {
  test('should render correctly and snapshot', () => {
    render(<SearchInput onChangeSearch={() => {}} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  test('Validate call onChangeSearch', () => {
    const onChangeSearch = jest.fn();
    render(<SearchInput onChangeSearch={onChangeSearch} />);
    const input = screen.getByTestId('search-input');
    fireEvent.changeText(input, 'test');
    expect(onChangeSearch).toBeCalledTimes(1);
  });
});
