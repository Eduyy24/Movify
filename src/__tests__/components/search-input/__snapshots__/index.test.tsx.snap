// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> should render correctly and snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 60,
      "paddingHorizontal": 10,
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    style={
      {
        "borderBottomColor": "#DBB4AD",
        "borderBottomWidth": 1,
        "flex": 1,
        "height": 40,
      }
    }
    testID="search-input"
  />
  <Image
    source={
      {
        "testUri": "../../../src/assets/icons/search.png",
      }
    }
  />
</View>
`;
